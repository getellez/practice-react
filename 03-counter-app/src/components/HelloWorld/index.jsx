import React from 'react'

export const HelloWorld = () => {
  const newMessage = {
    author: 'German',
    date: new Date(),
    categories: [1,2,3,4,5]
  }
  const getResult = () => {
    return 2+2;
  }
  return (
    <>
      <h1>Este es mi primer componente</h1>
    </>
  )
}
