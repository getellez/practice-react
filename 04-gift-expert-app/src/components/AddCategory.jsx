import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({ handleAddCategory }) => {
  const [inputValue, setInputValue] = useState('')
  
  const onInputChange = ({ target }) => {
    setInputValue(target.value)
  }
  const onSubmit = (event) => {
    event.preventDefault();
    if(inputValue.trim().length <= 1) return
    handleAddCategory(inputValue.trim())
    setInputValue('')
  }
  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text" 
        placeholder="Buscar GIFs"
        name="categoryInput"
        autoComplete="off"
        value={inputValue}
        onChange={onInputChange}
        />
    </form>
  )
}

AddCategory.propTypes = {
  handleAddCategory: PropTypes.func.isRequired,
}