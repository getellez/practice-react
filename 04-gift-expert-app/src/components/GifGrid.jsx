import { useFetchGifs } from "../hooks/useFetchGifs"
import { GifItem } from "./GifItem"
import PropTypes from 'prop-types'

export const GifGrid = ({ category, handleRemoveCategory }) => {
  
  const { images, isLoading } = useFetchGifs(category)
  
  return (
    <>
      <h3> { category } </h3> 
      {/* <small>[Eliminar]</small> */}
      <small onClick={() => handleRemoveCategory(category)}>[Eliminar]</small>
      
      {
        isLoading && (<h2>Loading...</h2>)
      }
      <div className="card-grid">
        {
          images.map(({ url, title, id }) => (
            <GifItem key={id} url={url} title={title} />
          ))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
  handleRemoveCategory: PropTypes.func.isRequired
}