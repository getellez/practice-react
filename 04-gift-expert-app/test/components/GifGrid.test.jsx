const { render, screen } = require("@testing-library/react");
import { GifGrid } from '../../src/components/GifGrid';
import { useFetchGifs } from '../../src/hooks/useFetchGifs';

jest.mock('../../src/hooks/useFetchGifs')

describe('<GifGrid /> component', () => {
  const category = 'One Punch Man'
  const handleRemoveCategoryMockFn = jest.fn()

  test('should match the snapshot when initial state', () => {
    useFetchGifs.mockReturnValue({
      images: [],
      isLoading: true
    })
    const { container } = render(<GifGrid category={category} handleRemoveCategory={handleRemoveCategoryMockFn}/>)
    expect(container).toMatchSnapshot()
  });
  test('should show loading message and category name', () => {
    useFetchGifs.mockReturnValue({
      images: [],
      isLoading: true
    })
    render(<GifGrid category={category} handleRemoveCategory={handleRemoveCategoryMockFn} />)
    expect(screen.getByText('Loading...'))
    expect(screen.getByText(category))
  });

  test('should show items when load images from useFetchGifs ', () => {
    const gifs = [ 
      {
        id:'123',
        title: 'Saitama',
        url:'https://localhost/saitama'
      },
      {
        id:'1234',
        title: 'Saitama 2',
        url:'https://localhost/saitama-2'
      },
    ]
    useFetchGifs.mockReturnValue({
      images: gifs,
      isLoading: false
    })

    render(<GifGrid category={category} handleRemoveCategory={handleRemoveCategoryMockFn} />)
    expect(screen.getAllByRole('img').length).toBe(2)
    expect(screen.getByText(gifs[0].title))
    expect(screen.getByText(gifs[1].title))
  });
});