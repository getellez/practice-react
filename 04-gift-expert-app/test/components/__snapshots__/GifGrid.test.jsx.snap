// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> component should match the snapshot when initial state 1`] = `
<div>
  <h3>
     
    One Punch Man
     
  </h3>
  <small>
    [Eliminar]
  </small>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
